### ISSUE TITLE
Indentation is not preserved when pasting from Word

### ISSUE THREAD

**Issue by Zonciu** [ckeditor/ckeditor-dev#651]
## Are you reporting a feature or a bug?
Bug

When I paste text from word, the paragraph indent is missing

### Expected result

Keep paragraph indent

### Actual result

No indent for paragraph

## Other details

* Browser: Chrome 59.0.3071.115
* OS: Windows 10
* CKEditor version: 4.7.1
* Installed CKEditor plugins: clipboard, pastefromword

**Comment by beatadelura** [comment]
@Zonciu Could you please provide more information about this issue, e.g. MS Word version, word file example with the text you're trying to paste and exact steps to reproduce?

**Comment by Zonciu** [comment]
@beatadelura 
Office pro plus 2016
I'd tried doc and docx both.

Word file:
[IndentTest.docx](https://github.com/ckeditor/ckeditor-dev/files/1155176/IndentTest.docx)


Here is the raw paste text I got
```
<div style="border: 1px solid;" contenteditable="true">

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">No indent</span></p>

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">&nbsp;</span></p>

<p style="margin: 0px; text-indent: 24pt;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">Right click – paragraph
– indent first line – 2 char</span></p>

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">&nbsp;</span></p>

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;"><span style="margin: 0px;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Indent with tab key</span></p>

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">&nbsp;</span></p>

<p style="margin: 0px 0px 0px 28px; text-indent: 24pt;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">Right click
– paragraph – indent first line – 2 char, and then indent with tab key</span></p>

<p style="margin: 0px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">&nbsp;</span></p>

<p style="margin: 0px 0px 0px 28px;"><span lang="EN-US" style="margin: 0px; font-size: 12pt;">Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph. Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph. Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph. Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph. Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph. Indent the whole paragraph.
Indent the whole paragraph. Indent the whole paragraph.</span></p>

</div>
```

In pastfromword plugin's filter, normalizedStyles function removes 'text-indent' style from the raw text
https://github.com/ckeditor/ckeditor-dev/blob/master/plugins/pastefromword/filter/default.js#L501

If I set the indent by righ click - paragraph - indent, the indent will be remove,
If I set the indent by tab key, the indent will not be remove —— because the indent was transform to space, not a `text-indent` style anymore.

If I make it to ignore the pastfromword filter, the raw text will be pasted and it shows very good, but I'm not sure will it cause bugs when I edit it in ckeditor.

**Comment by beatadelura** [comment]
@Zonciu Thank you for details. In CKEditor, we support margins for text indentation, so `text-indent` is treated as a redundant. Anyway, It's a good idea to extend support for `pastefromword` plugin, so I'll treat this issue as a feature request.

**Comment by celsowm** [comment]
Hi @beatadelura any news about this feature? thanks in advance !

**Comment by fikoborquez** [comment]
@beatadelura please let us know if there is any news

**Comment by mlewand** [comment]
@celsowm @fikoborquez please add 👍 reaction to the first comment in this issue. This is the best way to show us your support for this feature request.

**Comment by engineering-this** [comment]
In case with paragraph, this feature is simply to implement. Word outputs paragraph indention as left margin and first line indention as `text-indent` which is relative to margin. As we don't support `text-indention` style in our editor, filtering this attribute doesn't break any things, and everything looks pretty ok.

However in case of lists there are some difficulties.

![screen shot 2018-06-19 at 13 07 12](https://user-images.githubusercontent.com/23366849/41593904-e1b2ed5e-73c1-11e8-8c69-a812397768ea.png)

Second slider outputs `margin-left`, in this case it is 0.5 inch.
First slider outputs `text-indent` in this case it is -0.25 inch.

Currently we ignore both values for lists.

We could calculate proper margin value which would be sum of both values, but that would change how all lists are pasted and if user didn't change any indents in word it would be strange experience to have margins added. That also leaves us with question, new lists without any styling changed in word, after pasting to editor would have margin, and new list created in our editor wouldn't. A bit unexpected behaviour.

Other idea would be to reduce margin by default value, but that would be problematic. If user removed indent on list we would have to detect that, and handle it in specific way. Also there are possible complications with nested lists. 

We could consider bigger feature in future which would be adding word-like ruler with sliders, and allowing `text-indent` in Editor. In case of lists we should use `padding-left` instead, because text indent moves text not bullet.

For now I will propose PR with only paragraphs affected, lists will act the same way as now. Changing lists would make this feature much more complex, and plenty of cases should be considered.

**Comment by Comandeer** [comment]
This feature was merged into `major`, will be available in CKEditor 4.11.0.

**Comment by mlewand** [comment]
Initial report touched two separate topics. Margins and text indentation. Though both looks like "text spacing thingy", the two are different [as explained by @engineering-this](https://github.com/ckeditor/ckeditor-dev/issues/651#issuecomment-398372621). Thus purpose of this ticket was clarified to cover paragraphs margins.

Text indentation has been extracted to #2198 feature request.

I have updated the original report to reflect that.

And for all of you watching this issue you can simply try it out [in our major release (4.11.0) preview](https://ckeditor4-preview.ckeditor.com/major/samples/index.html).
